contracts/core/PollingPlace.sol

```diff
@@ -249,10 +235,14 @@ contract PollingPlace is PollingPlaceInterface {
 
         metaBlockGate = _metaBlockGate;
 
-        originCoreIdentifier = _originCoreIdentifier;
-        auxiliaryCoreIdentifier = _auxiliaryCoreIdentifier;
-        blockStores[originCoreIdentifier] = BlockStoreInterface(_originBlockStore);
-        blockStores[auxiliaryCoreIdentifier] = BlockStoreInterface(_auxiliaryBlockStore);
+        BlockStoreInterface originBlockStore = BlockStoreInterface(_originBlockStore);
+        BlockStoreInterface auxiliaryBlockStore = BlockStoreInterface(_auxiliaryBlockStore);
+
+        originCoreIdentifier = originBlockStore.coreIdentifier();
```

Comment:
Good question. I am not sure this is the right approach. I was thinking it is annoying that we always have to do *two* calls to get the core identifier for the auxiliary block store. Whenever we want to check a vote and where it belongs to. First to origin, realize it is not the same, then get it from auxiliary and compare. This way, we can have easy map access:
```solidity
/** The core identifier of the core that tracks origin. */
bytes20 public originCoreIdentifier;

/** The core identifier of the core that tracks this auxiliary chain. */
bytes20 public auxiliaryCoreIdentifier;
```

But like I said I am also not sure.

Reply:
What do you think @sarvesh-ost / @deepesh-kn ?

Reply:
I think getting data from another contract will consume more gas as compared to reading it from the variable. In past, I have seen that we have stored the value in the variables like this.
Storing is one time process here, but reading will be multiple time, so we should evaluate accordingly. My personal preference will be to keep it like this if gas is saved.

Reply:
If we duplicate the data, every ethereum node in the world needs to store that information. 
But here data is not that significantly big. 

I am not sure what is correct.  

@benjaminbollen  what do you think?

Reply:
The variable must be loaded from storage in both cases. The difference is storing vs. calling.

Storing the two variables costs `2 * 20,000 = 40,000` gas once.
Calling on the other contracts costs `1.5 * 2,000 = 3,000` gas per block.
(on average 1.5 calls as we don't know which blockstore of the two we need to call)

This means it is cheaper to store it starting from block `40,000 / 3,000 = 14`.
Assuming 10 seconds per block, **it would take less than 3 minutes to be cheaper to store than to call.**

Seeing that the code is clearer and it is cheaper when storing the variables, I would opt for storing.

Is my calculation correct?

I used the code below to compare gas costs of reading:
```solidity
pragma solidity ^0.4.23;

contract Returns {
    bytes20 public a;
    Stores public stores;
    
    constructor (bytes20 _val) public {
        stores = new Stores(_val);
        a = _val;
    }
    
    function remote() external view returns (bytes20) {
        return stores.b();
    }
}

contract Stores {
    bytes20 public b;
    
    constructor (bytes20 _val) public {
        b = _val;
    }
}
```